tests/components/sensor/test_fido.py

```diff
@@ -0,0 +1,112 @@
+"""The test for the fido sensor platform."""
+import asyncio
+import logging
+import sys
+from unittest.mock import MagicMock
+
+from homeassistant.bootstrap import async_setup_component
+from homeassistant.components.sensor import fido
+from tests.common import assert_setup_component
+
+
+CONTRACT = "123456789"
+
+
+class FidoClientMock():
+    """Fake Fido client."""
+
+    def __init__(self, username, password, contract=None):
+        """Fake Fido client init."""
+        pass
+
+    def get_phone_numbers(self):
+        """Return Phone numbers."""
+        return ["1112223344"]
+
+    def get_data(self):
+        """Return fake fido data."""
+        return {"balance": 160.12,
+                "1112223344": {"data_remaining": 100.33}}
+
+    @asyncio.coroutine
+    def fetch_data(self):
+        """Return fake fetching data."""
+        pass
+
+
+class FidoClientMockError(FidoClientMock):
+    """Fake Fido client error."""
+
+    @asyncio.coroutine
+    def fetch_data(self):
+        """Return fake fetching data."""
+        raise PyFidoErrorMock("Fake Error")
+
+
+class PyFidoErrorMock(BaseException):
```

Comment:
Minor comment, but according to the python docs `BaseException` shouldn't be inherited from. They suggest inheriting from `Exception`.
https://docs.python.org/3/library/exceptions.html